plugin/azure/azure.go

```diff
@@ -0,0 +1,227 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/file"
+	"github.com/coredns/coredns/plugin/pkg/fall"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+	"github.com/coredns/coredns/request"
+
+	azure "github.com/Azure/azure-sdk-for-go/profiles/latest/dns/mgmt/dns"
+	"github.com/miekg/dns"
+)
+
+type zone struct {
+	id  string
+	z   *file.Zone
+	dns string
```

Comment:
`dns` is oddly named, file.Zone also has a field origin that you mean to reuse that field? If it needs to be here (I think you can use file.Zone.origin) is should be named `name` or `origin`

Reply:
renamed it to `zone`